FirebaseVertexAI/Sources/Chat.swift

```diff
@@ -26,9 +26,29 @@ public class Chat {
     self.history = history
   }
 
+  private let historyLock = NSLock()
+  private var _history: [ModelContent] = []
   /// The previous content from the chat that has been successfully sent and received from the
   /// model. This will be provided to the model for each message sent as context for the discussion.
-  public var history: [ModelContent]
+  public var history: [ModelContent] {
+    get {
+      historyLock.lock()
+      let result = _history
+      historyLock.unlock()
+      return result
+    }
+    set {
+      historyLock.lock()
+      _history = newValue
+      historyLock.unlock()
+    }
```

Comment:
nit: Thoughts on this syntax instead? (Feel free to ignore, should be functionally the same.)

```suggestion
    get {
      historyLock.withLock { _history }
    }
    set {
      historyLock.withLock { _history = newValue }
    }
```

Reply:
Definitely cleaner. Thanks!